### ISSUE TITLE
Typed or Copied Content Replaces Custom Alert Block Instead of Staying Within It

### ISSUE THREAD

**Issue by lamtuanamc** [TypeCellOS/BlockNote#771]
Description:
I have customized a block type "alert" followed by a custom block on the BlockNote homepage. However, every time I type '-', '*', '+', '1.' or copy a text segment within the same editor, it replaces the "alert" block instead of placing the content within it.

Steps to Reproduce:

Create a custom "alert" block as per the documentation.
Type '-', '*', '+', '1.' or copy and paste a text segment within the "alert" block.
Observe that the content replaces the "alert" block instead of staying within it.
Expected Behavior:
The typed or copied content should remain within the "alert" block instead of replacing it.

Actual Behavior:
The "alert" block is replaced by the typed or copied content.

Link to Documentation:
[BlockNote Custom Schemas - Custom Blocks](https://www.blocknotejs.org/docs/custom-schemas/custom-blocks)

**Comment by YousefED** [comment]
I agree this is undesirable. It should probably be fixed here: https://github.com/TypeCellOS/BlockNote/blob/bea361e39846c26d9729a777a5cd3de1f48e8c3e/packages/core/src/blocks/ListItemBlockContent/BulletListItemBlockContent/BulletListItemBlockContent.ts#L49 (and in the code for related shortcuts)

**Comment by lamtuanamc** [comment]
@YousefED If I remove this part, won't the regular text lines be unable to trigger the creation of a list? Or is there a way to check so that other normal blocks are not affected?

**Comment by matthewlipski** [comment]
You might want to call `getBlockInfoFromPos` instead and check if the block type is part of the default blocks:

```ts
const { contentType } = getBlockInfoFromPos(
  this.editor.state.doc,
  this.editor.state.selection.anchor
);

if (contentType.name in defaultBlockSpecs) {
  return true;
}
```
